File: arsenal/sleep/aws/csharp/Handler.cs
using System.Collections.Generic;
using System.Threading;
using Amazon.Lambda.Core;
using System;

[assembly:LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]

namespace AwsDotnetCsharp
{
    public class Handler
    {
       public int GetSleepParameter(Dictionary<String, String> input){
          String user_input = input["sleep"];
          if (user_input != null) {
             int retval = Int32.Parse(user_input);
	     return retval > -1 ? retval : -1;
	  }
	  return -1;
       }

       public int GetParameters(Dictionary<String, String> input){
            return GetSleepParameter(input);
       }

       public void RunTest(int sleepTime) {
	    Thread.Sleep(sleepTime);
       }

       public bool isWarm() {
            bool result = System.Environment.GetEnvironmentVariable("warm") == "true";
            System.Environment.SetEnvironmentVariable("warm", "true");
            return result;
       }


       public Response Hello(Dictionary<string, string> request)
       {
	   long start = new DateTimeOffset(DateTime.UtcNow).ToUnixTimeMilliseconds();
	   int sleepTime = GetParameters(request);
	   RunTest(sleepTime);
	   bool reused = isWarm();
	   long end = new DateTimeOffset(DateTime.UtcNow).ToUnixTimeMilliseconds();
	   long duration = (end - start)*1000000;
           return new Response(duration, reused);
       }
    }

    public class Response
    {
       public long duration {get; set;}
       public bool reused {get; set;}

       public Response(long duration, bool reused){
	   this.reused = reused;
	   this.duration = duration;
       }
    }
}

File: arsenal/sleep/aws/python/handler.py
import json
import os
import time
import datetime

def get_current_epoch():
    return int((datetime.datetime.utcnow() - datetime.datetime(1970, 1, 1)).total_seconds() * 1000)

def get_sleep_parameter(event):
    user_input = str(event.get("sleep"))
    if not user_input or not user_input.isdigit() or int(user_input) < 0:
        return {"error": "invalid sleep parameter"}
    return int(user_input) 

def get_parameters(event):
    return get_sleep_parameter(event)

def run_test(sleep_time):
    time.sleep(sleep_time / 1000.0)

def is_warm():
    is_warm = os.environ.get("warm") == "true"
    os.environ["warm"] = "true"
    return is_warm

def hello(event, context):
    start = get_current_epoch()
    reused = is_warm()
    sleep_time = get_parameters(event)
    if type(sleep_time) != int:
        return sleep_time
    run_test(sleep_time)
    duration = (get_current_epoch() - start) * 1000000

    return {
        "duration": duration,
        "reused": reused
    }
